### DISCUSSION TITLE
Failed to run sql query: query returned no rows

### DISCUSSION THREAD

**Discussion by sumatoken** [supabase/supabase#16633] [Questions]
I have a project on Supabase and I've wanted to implement TCE to encrypt a certain column.

I am running into errors when trying to insert into my table which has an encrypted column.

Here is my table definition:
```sql
create table
  public.user_databases_tce (
    id bigint generated by default as identity,
    created_at timestamp with time zone null default now(),
    database_string text null,
    user_id text null default requesting_user_id (),
    uuid uuid not null,
    show_sample boolean null default false,
    constraint user_databases_pkey_tce primary key (id, uuid),
    constraint user_databases_uuid_key_tce unique (uuid),
    constraint user_databases_uuid_fkey foreign key (uuid) references user_schemas (uuid) on delete cascade
  ) tablespace pg_default;

create trigger user_databases_tce_encrypt_secret_trigger_database_string before insert
or
update of database_string on user_databases_tce for each row
execute function user_databases_tce_encrypt_secret_database_string ();
```
And here is how I activated column encryption for the *database_string* column:

```sql
SECURITY LABEL FOR pgsodium
	ON COLUMN user_databases.database_string
  	IS 'ENCRYPT WITH KEY COLUMN uuid';
```

FYI, the **user_databases_tce_encrypt_secret_trigger_database_string** function is defined automatically after I run the security label query:

```sql
CREATE OR REPLACE FUNCTION 
public.user_databases_tce_encrypt_secret_database_string()
 RETURNS trigger
 LANGUAGE plpgsql
AS $function$
        BEGIN
                new.database_string = CASE WHEN new.database_string IS NULL THEN NULL ELSE
            CASE WHEN new.uuid IS NULL THEN NULL ELSE pg_catalog.encode(
              pgsodium.crypto_aead_det_encrypt(
                pg_catalog.convert_to(new.database_string, 'utf8'),
                pg_catalog.convert_to(('')::text, 'utf8'),
                new.uuid::uuid,
                NULL
              ),
                'base64') END END;
        RETURN new;
        END;
        $function$
```

Now, when I try to insert in the table using the JavaScript Supabase client or even the Supabase dashboard GUI, I get this error:
```json
{
    "error": {
        "code": "P0002",
        "details": null,
        "hint": null,
        "message": "query returned no rows"
    }
}
```

I don't know if I'm doing something wrong or if I need to configure some type of policy.

I'd appreciate your guys' help.

**Comment by seho0808** [comment]
Do you know if it has any RLS (Row Level Security) enabled?
I had many cases where I see nothing returned and it was due to RLS.

EDIT: if its from GUI it should bypass RLS... unless you changed the roles provided by supabase

**Reply by sumatoken** [reply]
I have RLS disabled on the table in question. And Yes, it shouldn't be an RLS issue since I'm inserting it from the GUI.

**Comment by GaryAustin1** [comment]
Your error is in your Case code most likely.  The SQL error is saying some part of that should be returning a result and it is not getting one.
I'm not good enough at SQL to quickly spot it.  You might debug the case statement in standalone SQL first or even ask chatGPT.

**Comment by devbydaniel** [comment]
I receive the same error after encrypting a certain column in a table. I did the encryption via the GUI, then pulled the new DB schema to my local project. The table definition looks exactly like OP's

As a result, I receive the same error when trying to insert data in my local project (both via GUI and via edge function with service role), however I *am* able to insert data in my remote project. Any ideas?

**Comment by anthonyverducci** [comment]
Hi all, I opened a bug report for this issue: #18192

**Comment by jonkurtis** [comment]
You need to set a default uuid value for the encryption to work. Add this:
```
DO $$
DECLARE
    key_id_value uuid;
BEGIN

  SELECT id INTO key_id_value FROM pgsodium.create_key(name := 'default_secrets_keys');

  EXECUTE 'ALTER TABLE public.user_databases_tce ALTER COLUMN uuid SET DEFAULT ' || quote_literal(key_id_value) || '::uuid';

END;
$$ LANGUAGE plpgsql;
```

**Reply by jonkurtis** [reply]
Alternatively, you sould run:
`select * from pgsodium.create_key();`
and then go set the default value of the uuid column with the id returned from that command in the GUI.
